### DISCUSSION TITLE
Allow configuring custom capabilities (e.g. to disable snippets in completions)

### DISCUSSION THREAD

**Discussion by bew** [mrcjkb/rustaceanvim#545] [Q&A]
### Feature description

Hello !
I'm considering using this plugin, but looking at [how capabilities are defined](https://github.com/mrcjkb/rustaceanvim/blob/4e78e8d8fe464fbc9e99632b65dd3ead75d9dda0/lua/rustaceanvim/config/server.lua#L153) it looks like there are no way to pass custom capabilities..

In my previous config I used:
```lua
    capabilities = vim.tbl_deep_extend(
      "force",
      vim.lsp.protocol.make_client_capabilities(),
      require"cmp_nvim_lsp".default_capabilities(),
      {
        -- Disable snippets in completion candidates
        textDocument = { completion = { completionItem = { snippetSupport = false } } }
      }
    ),
```
I know this plugin automatically loads default caps + cmp_nvim_lsp's, but I don't think there's a way to force pass the last section to disable snippets in completions 🤔

**Comment by mrcjkb** [comment]
Hey :wave: 

Here's an example snippet:

```lua
vim.g.rustaceanvim = {
  server = {
    capabilities = {
      textDocument = {
        completion = {
          completionItem = {
            snippetSupport = false,
          },
        }
      },
    },
  },
}
```

See also `:h rustaceanvim.config`.
